### PR TITLE
Adding possibility to undock the panel chat widget

### LINKED ISSUE #197219: [Feature] Undock view into separate editor
We were discussing the panel chat with @isidorn, and I said we could perhaps have a command to undock the copilot chat view. I think this is currently possible by placing the view into an editor and then undocking the editor, but could we perhaps have a command which would directly do that? I think it would be useful in the case of copilot-chat because the panel chat takes up quite a lot of space.

### PR DESCRIPTION
**Pull request by aiday-mar**
Fixes https://github.com/microsoft/vscode/issues/197219

This PR is inspired by the work previously done by @roblourens. There is already an action both in the command palette and in the chat menu to move the chat session to a new editor. This PR extends the work to be able to move the chat session to a new window. The PR also includes some refactoring. 

https://github.com/microsoft/vscode/assets/61460952/1a43e1fd-06f3-44b7-8914-db8c248a9066

### CODE REVIEW

File: src/vs/workbench/contrib/chat/browser/chat.contribution.ts [hunk 1]
```diff
@@ -234,6 +238,50 @@ class ChatSlashStaticSlashCommandsContribution extends Disposable {
...
+		const editorPanesRegistry = Registry.as<IEditorPaneRegistry>(EditorExtensions.EditorPane);
+		const instantiationService = accessor.get(IInstantiationService);
+		const chatEditorInput: ChatEditorInput = instantiationService.createInstance(ChatEditorInput, ChatEditorInput.getNewEditorUri(), { target: { sessionId } });
+		const editorPaneDescriptor = editorPanesRegistry.getEditorPane(chatEditorInput);
+		const auxiliaryEditorPart = await editorGroupService.createAuxiliaryEditorPart();
+		const editorPane = editorPaneDescriptor?.instantiate(instantiationService);
+		if (!editorPane || !editorPane.group || !editorPane.input) {
+			return;
+		}
+		editorPane.group.moveEditor(editorPane.input, auxiliaryEditorPart.activeGroup);
```
Comment (bpasero) [inline src/vs/workbench/contrib/chat/browser/chat.contribution.ts:281]: Hm I was not expecting a `moveEditor` but just `openEditor` in the aux window?

Comment (aiday-mar) [reply src/vs/workbench/contrib/chat/browser/chat.contribution.ts]: Hi @bpasero, it's for the moment a draft PR, the code currently does not work and I was planning to dig into why this is the case. But thanks for the tip, I'll have a look more closely at the `openEditor` method

Comment (bpasero) [reply src/vs/workbench/contrib/chat/browser/chat.contribution.ts]: Sure thing, let me know if you need help.

**Review by bpasero — APPROVED** [review]
🆒 . I see multiple locations (3?) of

```ts
const auxiliaryEditorPart = await editorGroupService.createAuxiliaryEditorPart();
				const chatEditorInput = instantiationService.createInstance(ChatEditorInput, ChatEditorInput.getNewEditorUri(), { target: { providerId } });
				auxiliaryEditorPart.activeGroup.openEditor(chatEditorInput, { pinned: true });
```

Maybe thats a candidate for having a reusable method with parameters for this purpose.

**Comment by aiday-mar** [comment]
Ah thank you for commenting about it, I can make a second PR for this.

**Pull request merged**